pkg/auth/api/api.go

```diff
@@ -393,6 +400,52 @@ func (s *authService) checkEmail(
 	return dt.Err()
 }
 
+// Check if the user has at least one account enabled in any Organization
+func (s *authService) checkAccountStatus(
+	ctx context.Context,
+	email string,
+	organizations []*envproto.Organization,
+	localizer locale.Localizer,
+) error {
+	for _, org := range organizations {
+		resp, err := s.accountClient.GetAccountV2(ctx, &acproto.GetAccountV2Request{
+			Email:          email,
+			OrganizationId: org.Id,
+		})
+		if err != nil {
+			if errors.Is(err, accountstotage.ErrAccountNotFound) {
+				s.logger.Error("Account not found",
+					zap.Error(err),
+					zap.String("email", email),
+					zap.String("organizationId", org.Id),
+				)
+				continue
+			}
+			dt, err := auth.StatusInternal.WithDetails(&errdetails.LocalizedMessage{
+				Locale:  localizer.GetLocale(),
+				Message: localizer.MustLocalize(locale.InternalServerError),
+			})
+			if err != nil {
+				return auth.StatusInternal.Err()
+			}
+			return dt.Err()
+		}
+		if !resp.Account.Disabled {
```

Comment:
If the user has at least one account enabled, we allow the access token generation.